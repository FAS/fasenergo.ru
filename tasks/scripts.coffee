module.exports = () ->

  ###
  Shell
  https://github.com/sindresorhus/grunt-shell
  Run shell commands
  ###

  @config 'shell',
    jspm_config:
      command: 'jspm config registries.github.auth <%= env.github.api.key %>'
    jspm_install:
      command: 'jspm install'
    jspm_build:
      command: 'jspm build <%= file.source.script %> <%= file.build.script.compiled %> --minify'

  ###
  Uglify
  https://github.com/gruntjs/grunt-contrib-uglify
  Minify files with UglifyJS
  ###

  @config 'uglify',
    build:
      files: [
        expand: true
        cwd: '<%= path.build.scripts %>'
        src: '{,**/}*.js'
        dest: '<%= path.build.scripts %>'
        ext: '.min.js'
      ]

  ###
  Clean
  https://github.com/gruntjs/grunt-contrib-clean
  Clean folders to start fresh
  ###

  @config.merge
    clean:
      scripts:
        files: [
          expand: true
          cwd: '<%= path.build.scripts %>'
          src: [
            '{,**/}*.*'
            '!{,**/}*.min.js'
          ]
        ]

  ###
  Standard
  https://github.com/pdehaan/grunt-standard
  Lint JS files with standard
  ###

  @config 'standard',
    lint:
      files: [
        expand: true
        cwd: '<%= path.source.scripts %>'
        src: '{,**/}*.js'
      ]
    format:
      options:
        format: true
      files: [
        expand: true
        cwd: '<%= path.source.scripts %>'
        src: '{,**/}*.js'
      ]

  ###
  Watch
  https://github.com/gruntjs/grunt-contrib-watch
  Watches scss, js etc for changes and compiles them
  ###

  @config.merge
    watch:
      scripts:
        files: ['<%= path.source.scripts %>/{,**/}*.js']
        tasks: ['shell:jspm_build']