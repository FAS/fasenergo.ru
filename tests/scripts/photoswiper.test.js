/**
 * @jest-environment jsdom
 */

/* eslint-env jest */

import '../../source/scripts/polyfill'
import photoSwiper, { getImages } from '../../source/scripts/plugins/photoswiper'

let activePhotoswipe

jest.mock('photoswipe', () => class PhotoSwipe {
  constructor ($element, ui, items, options) {
    this.items = items
    this.options = options
  }

  init () {
    activePhotoswipe = {
      index: this.options.index,
      items: this.items
    }
  }
}, { virtual: true })

jest.mock('photoswipe/dist/photoswipe-ui-default.min.js', () => {}, { virtual: true })
jest.mock('photoswipe/dist/photoswipe.css!', () => {}, { virtual: true })
jest.mock('photoswipe/dist/default-skin/default-skin.css!', () => {}, { virtual: true })

const Image = (thumbnail, image, width, height, alt, caption) => `
  <figure>
    <a class='js-photoswiper__image' href='${image}' data-width='${width}' data-height='${height}'>
      <img src='${thumbnail}' alt='${alt}' />
    </a>
    <figcaption>${caption}</figcaption>
  </figure>
`

describe('PhotoSwiper', () => {
  describe('`getImages()`', () => {
    it('should get images data', () => {
      document.body.innerHTML = `
        ${Image('/thumbnail-1.png', '/image-1.png', 100, 150, 'Thumbnail-1 alt', 'Thumbnail-1 caption')}
        ${Image('/thumbnail-2.png', '/image-2.png', 200, 250, 'Thumbnail-2 alt', 'Thumbnail-2 caption')}
      `

      expect(getImages(document.body, '.js-photoswiper__image')).toMatchSnapshot()
    })
  })

  it('should work with galleries', () => {
    document.body.innerHTML = `
      <article class='js-photoswiper'>
        ${Image('/thumbnail-1.png', '/image-1.png', 100, 150, 'Thumbnail-1 alt', 'Thumbnail-1 caption')}
        ${Image('/thumbnail-2.png', '/image-2.png', 200, 250, 'Thumbnail-2 alt', 'Thumbnail-2 caption')}
      </article>

      <article class='js-photoswiper'>
        ${Image('/thumbnail-3.png', '/image-3.png', 300, 350, 'Thumbnail-3 alt', 'Thumbnail-3 caption')}
        ${Image('/thumbnail-4.png', '/image-4.png', 400, 450, 'Thumbnail-4 alt', 'Thumbnail-4 caption')}
      </article>
    `

    photoSwiper()

    document.querySelector('[href="/image-1.png"]').click()
    expect(activePhotoswipe).toMatchSnapshot()

    document.querySelector('[href="/image-4.png"]').click()
    expect(activePhotoswipe).toMatchSnapshot()
  })
})
