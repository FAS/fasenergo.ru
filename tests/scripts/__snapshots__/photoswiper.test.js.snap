// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhotoSwiper \`getImages()\` should get images data 1`] = `
Array [
  Object {
    "$thumbnail": <img
      alt="Thumbnail-1 alt"
      src="/thumbnail-1.png"
    />,
    "h": 150,
    "msrc": "/thumbnail-1.png",
    "src": "/image-1.png",
    "title": "Thumbnail-1 caption",
    "w": 100,
  },
  Object {
    "$thumbnail": <img
      alt="Thumbnail-2 alt"
      src="/thumbnail-2.png"
    />,
    "h": 250,
    "msrc": "/thumbnail-2.png",
    "src": "/image-2.png",
    "title": "Thumbnail-2 caption",
    "w": 200,
  },
]
`;

exports[`PhotoSwiper should work with galleries 1`] = `
Object {
  "index": 0,
  "items": Array [
    Object {
      "$thumbnail": <img
        alt="Thumbnail-1 alt"
        src="/thumbnail-1.png"
      />,
      "h": 150,
      "msrc": "/thumbnail-1.png",
      "src": "/image-1.png",
      "title": "Thumbnail-1 caption",
      "w": 100,
    },
    Object {
      "$thumbnail": <img
        alt="Thumbnail-2 alt"
        src="/thumbnail-2.png"
      />,
      "h": 250,
      "msrc": "/thumbnail-2.png",
      "src": "/image-2.png",
      "title": "Thumbnail-2 caption",
      "w": 200,
    },
  ],
}
`;

exports[`PhotoSwiper should work with galleries 2`] = `
Object {
  "index": 1,
  "items": Array [
    Object {
      "$thumbnail": <img
        alt="Thumbnail-3 alt"
        src="/thumbnail-3.png"
      />,
      "h": 350,
      "msrc": "/thumbnail-3.png",
      "src": "/image-3.png",
      "title": "Thumbnail-3 caption",
      "w": 300,
    },
    Object {
      "$thumbnail": <img
        alt="Thumbnail-4 alt"
        src="/thumbnail-4.png"
      />,
      "h": 450,
      "msrc": "/thumbnail-4.png",
      "src": "/image-4.png",
      "title": "Thumbnail-4 caption",
      "w": 400,
    },
  ],
}
`;
